core/src/main/java/nu/yona/server/goals/service/GoalDTO.java

```diff
@@ -28,20 +30,21 @@
 {
 	private final UUID id;
 	private UUID activityCategoryID;
-	private final ZonedDateTime creationTime;
+	private final Optional<ZonedDateTime> creationTime;
 	private final boolean mandatory;
 
-	protected GoalDTO(UUID id, UUID activityCategoryID, ZonedDateTime creationTime, boolean mandatory)
+	protected GoalDTO(UUID id, Optional<ZonedDateTime> creationTime, UUID activityCategoryID, boolean mandatory)
 	{
+		Objects.requireNonNull(creationTime);
```

Comment:
That's correct, but currently we don't have a setting like that. The check would need to happen in the ``JsonCreator`` constructors of ``TimeZoneGoal`` and ``BudgetGoal``. Given that nothing really goes wrong if someone would squeeze the creation time in, I don't think it's worth the effort.